### DISCUSSION TITLE
Should the CancellationToken be a seperate parameter or should be a property on the FetchInfo

### DISCUSSION THREAD

**Discussion by inforithmics** [Mapsui/Mapsui#2548] [Development]
In this Pullrequest https://github.com/Mapsui/Mapsui/pull/2541 I add the CancellationToken to certain operations that run async Code to abort them.

There are now 2 possibilities to add this to the

1. Add it as a Property to FetchInfo
```
var fetchInfo = new FetchInfo(Navigator.Viewport.ToSection(), CRS, changeType, cancellationToken);
RefreshData(fetchInfo);
```

Pro: No interface change
Contra: less visible that this method supports a CancellationToken.

2. Add it as a separate Parameter

```
var fetchInfo = new FetchInfo(Navigator.Viewport.ToSection(), CRS, changeType);
RefreshData(fetchInfo,  cancellationToken);
```
Pro: immediately visible that it supports CancellationToken
Contra: Interface is changed so implementators need to change it.

Currently I have implemented option 1: Because it breaks less Code. I only wanted to ask what's the general opinion about this. I don't have a strong opinion about it. In the dotnet framework option 2 is taken, and the compiler can issue warnings when a cancellationtoken is not given to called methods.

**Comment by pauldendulk** [comment]
RefreshData is something the users calls so I don't think they should be bothered with a CancellationToken. But I don't think FetchInfo should have one either. Rather I was thinking this should be encapsulated by the MapControl or Map. Let's discuss this further in the issue.
